id: PYSEC-0000-CVE-2021-4041
details: A flaw was found in ansible-runner. An improper escaping of the shell command,
  while calling the ansible_runner.interface.run_command, can lead to parameters getting
  executed as host's shell command. A developer could unintentionally write code that
  gets executed in the host rather than the virtual environment.
affected:
- package:
    name: ansible-runner
    ecosystem: PyPI
    purl: pkg:pypi/ansible-runner
  ranges:
  - type: GIT
    repo: https://github.com/ansible/ansible-runner
    events:
    - introduced: "0"
    - fixed: 3533f265f4349a3f2a0283158cd01b59a6bbc7bd
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 2.1.0
references:
- type: REPORT
  url: https://bugzilla.redhat.com/show_bug.cgi?id=2028074
- type: WEB
  url: https://access.redhat.com/security/cve/CVE-2021-4041
- type: FIX
  url: https://github.com/ansible/ansible-runner/commit/3533f265f4349a3f2a0283158cd01b59a6bbc7bd
aliases:
- CVE-2021-4041
modified: "2022-08-29T14:30:00Z"
published: "2022-08-24T16:15:00Z"
